src/app/user/user-details/user-details.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { map, Observable, of, Subject, switchMap, takeWhile, tap } from 'rxjs';
+import { User } from '../../shared/models/user.model';
+import { UserService } from '../../shared/services/user.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { catchError } from 'rxjs';
+import { AsyncPipe } from '@angular/common';
+import { MatIconModule } from '@angular/material/icon';
+import { MatButtonModule } from '@angular/material/button';
+import { MatTooltipModule } from '@angular/material/tooltip';
+import { ReactiveFormsModule, FormControl, FormGroup } from '@angular/forms';
+import {MatSelectModule} from '@angular/material/select';
+
+@Component({
+  selector: 'user-details',
+  standalone: true,
+  imports: [AsyncPipe, MatSelectModule, MatIconModule, MatButtonModule, MatTooltipModule, ReactiveFormsModule],
+  templateUrl: './user-details.component.html',
+  styleUrl: './user-details.component.scss'
+})
+export class UserDetailsComponent implements OnInit, OnDestroy {
+  user$: Observable<User> | null = null;
+  private destroy$: Subject<void> = new Subject<void>();
+  editable: boolean = false;
+
+  roleForm = new FormGroup({
+    roles: new FormControl([''])
+  });
+
+  constructor(private userService: UserService, private route: ActivatedRoute, private router: Router) {}
+
+  ngOnInit(): void {
+    this.user$ = this.route.paramMap.pipe(
+      map(params => params.get("id")),
+      switchMap(id => {
+        if (id) {
+          return this.userService.getUser(id).pipe(
+            catchError(err => {
+              this.router.navigate(['/404']);
+              
+              return of();
+            })
+          );
+        } else {
+          this.router.navigate(['/404']);
+        
+          return of();
+        }
+      })
+    )
+  }
+
+  onEdit(roles: string[]) {
+    this.roleForm.setValue({
```

Comment:
In this case if you load the page and let's say user has both User and Manager roles, does that mean both options will be unchecked UNTIL you press edit? If so then please change to show the correct state all the time